raiden-cli/src/routes/channels.ts

```diff
@@ -52,7 +64,91 @@ async function openChannel(this: Cli, request: Request, response: Response, next
   } catch (error) {
     if (isInvalidParameterError(error)) {
       response.status(400).send(error.message);
-    } else if (error.code === 'INSUFFICIENT_FUNDS') {
+    } else if (isInvalidParameterError(error)) {
+      response.status(402).send(error.message);
+    } else if (isConflictError(error)) {
+      response.status(409).send(error.message);
+    } else {
+      next(error);
+    }
+  }
+}
+
+const allowedUpdateKeys = new Set(['state', 'total_deposit', 'total_withdraw']);
+function validateChannelUpdateBody(request: Request, response: Response, next: NextFunction) {
+  const intersec = Object.keys(request.body).filter((k) => allowedUpdateKeys.has(k));
+  if (intersec.length < 1)
+    return response
+      .status(400)
+      .send('one of "state" | "total_deposit" | "total_withdraw" operations required');
+  else if (intersec.length > 1)
+    return response
+      .status(409)
+      .send('more than one of "state" | "total_deposit" | "total_withdraw" requested');
+  if (
+    request.body.state &&
+    ![ApiChannelState.closed, ApiChannelState.settled].includes(request.body.state)
+  )
+    return response
+      .status(400)
+      .send('invalid "state" requested: must be one of "closed" | "settled"');
+  return next();
+}
+
+async function updateChannel(this: Cli, request: Request, response: Response, next: NextFunction) {
+  const token: string = request.params.tokenAddress;
+  const partner: string = request.params.partnerAddress;
+  try {
+    let channel = await this.raiden.channels$.pipe(first(), pluck(token, partner)).toPromise();
+    if (!channel) return response.status(404).send('channel not found');
+    if (request.body.state) {
+      if (request.body.state === ApiChannelState.closed) {
+        await this.raiden.closeChannel(token, partner);
+        channel = await this.raiden.channels$
+          .pipe(
+            pluck(token, partner),
+            first((channel) =>
+              [ChannelState.closed, ChannelState.settleable, ChannelState.settling].includes(
+                channel.state,
+              ),
+            ),
+          )
+          .toPromise();
+      } else if (request.body.state === ApiChannelState.settled) {
+        const promise = this.raiden.settleChannel(token, partner);
+        channel = await this.raiden.channels$.pipe(pluck(token, partner), first()).toPromise();
+        await promise;
+      }
+    } else if (request.body.total_deposit) {
+      await this.raiden.depositChannel(
+        token,
+        partner,
+        channel.ownDeposit.sub(request.body.total_deposit).mul(-1),
```

Comment:
Wait what. Why? Please add a comment if code can't express it. I think this is an important detail.
(same goes for the withdrawl below)

Reply:
I'll try to do that in code: SDK approach deals with deposit and withdraw amounts, not with totals. This here calculates that by e.g. subtracting `ownDeposit` from `total_deposit`, but since `ownDeposit` is already a `BigNumber`, and `total_deposit` is (expectedly) a `BigNumberish`, it's simpler to subtract in the reverse order and invert then, to avoid importing `bigNumberify` only for a call which there's already a method for.